kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/storageApi/StorageWriteApiBatchApplicationStream.java

```diff
@@ -0,0 +1,302 @@
+            throw new BigQueryConnectException(e);
+        }
+    }
+
+    /**
+     * Gets commitable offsets on all tables and all streams. Offsets returned should be sequential. As soon as we see a
+     * stream not committed we will drop iterating over next streams for that table. Cleans up committed streams
+     *
+     * @return Returns Map of TopicPartition to OffsetMetadata. Will be empty if there is nothing new to commit.
+     */
+    @Override
+    public Map<TopicPartition, OffsetAndMetadata> getCommitableOffsets() {
+        Map<TopicPartition, OffsetAndMetadata> offsetsReadyForCommits = new ConcurrentHashMap<>();
+        this.streams.forEach((tableName, streamDetails) -> {
+                    synchronized (lock(tableName)) {
+                        int i = 0;
+                        Set<String> deletableStreams = new HashSet<>();
+                        for (Map.Entry<String, ApplicationStream> applicationStreamEntry : streamDetails.entrySet()) {
+                            ApplicationStream applicationStream = applicationStreamEntry.getValue();
+                            String streamName = applicationStreamEntry.getKey();
+                            if (applicationStream.isInactive()) {
+                                logger.trace("Ignoring inactive stream {} at index {}...", streamName, i);
+                            } else if (applicationStream.isReadyForOffsetCommit()) {
+                                logger.trace("Pulling offsets from committed stream {} at index {} ...", streamName, i);
+                                offsetsReadyForCommits.putAll(applicationStream.getOffsetInformation());
+                                applicationStream.markInactive();
+                            } else {
+                                logger.trace("Ignoring all streams as stream {} at index {} is not yet committed", streamName, i);
+                                // We move sequentially for offset commit, until current offsets are ready, we cannot commit next.
+                                break;
+                            }
+                            deletableStreams.add(streamName);
+                            i++;
+                        }
+                        deletableStreams.forEach(streamDetails::remove);
+                    }
+                }
+        );
+
+        logger.trace("Commitable offsets are {} for all tables on all eligible stream  : ", offsetsReadyForCommits);
+
+        return offsetsReadyForCommits;
+    }
+
+
+    /**
+     * This attempts to create stream if there are no existing stream for table or the stream is not empty
+     * (it has been assigned some records)
+     *
+     * @param tableName Name of the table in project/dataset/table format
+     * @return
+     */
+    @Override
+    public boolean mayBeCreateStream(String tableName, List<Object[]> rows) {
+        String streamName = this.currentStreams.get(tableName);
+        boolean result = (streamName == null) ||
```

Comment:
```suggestion
        boolean isCreateNewStream = (streamName == null) ||
```

Reply:
`shouldCreateNewStream` ? @aniketshrimal

Reply:
Yes, that sounds better!